apps/insights/src/server/pyth.ts

```diff
@@ -0,0 +1,71 @@
+import { parse, } from "superjson";
+import { z } from "zod";
+
+import { PUBLIC_URL, VERCEL_AUTOMATION_BYPASS_SECRET } from '../config/server';
+import { Cluster, priceFeedsSchema } from "../services/pyth";
+import { DEFAULT_CACHE_TTL } from "../utils/cache";
+
+// Convenience helpers matching your previous functions
+export async function getPublishersForFeedCached(
+  cluster: Cluster,
+  symbol: string
+) {
+  const data = await fetch(`${PUBLIC_URL}/api/pyth/get-publishers/${encodeURIComponent(symbol)}?cluster=${cluster.toString()}`, {
+    next: {
+      revalidate: DEFAULT_CACHE_TTL,
+    },
+    headers: {
+      'x-vercel-protection-bypass': VERCEL_AUTOMATION_BYPASS_SECRET,
+    },
+  });
+  return data.json() as Promise<string[]>;
+}
+
+export async function getFeedsForPublisherCached(
+  cluster: Cluster,
+  publisher: string
+) {
+  const data = await fetch(`${PUBLIC_URL}/api/pyth/get-feeds-for-publisher/${encodeURIComponent(publisher)}?cluster=${cluster.toString()}`, {
+    next: {
+      revalidate: DEFAULT_CACHE_TTL,
+    },
+    headers: {
+      'x-vercel-protection-bypass': VERCEL_AUTOMATION_BYPASS_SECRET,
+    },
+  });
+  const rawData = await data.text();
+  return parse<z.infer<typeof priceFeedsSchema>>(rawData);
+}
+
+export const getFeedsCached = async (cluster: Cluster) => {
+  const data = await fetch(`${PUBLIC_URL}/api/pyth/get-feeds?cluster=${cluster.toString()}&excludePriceComponents=true`, {
+    next: {
+      revalidate: DEFAULT_CACHE_TTL,
+    },
+    headers: {
+      'x-vercel-protection-bypass': VERCEL_AUTOMATION_BYPASS_SECRET,
+    },
+  });
+  const dataJson = await data.text();
+  const feeds: Omit<z.infer<typeof priceFeedsSchema>[0], "price">[] = parse(dataJson);
+  return feeds;
+}
+
+export const getFeedForSymbolCached = async ({symbol, cluster = Cluster.Pythnet}: {symbol: string, cluster?: Cluster}): Promise<z.infer<typeof priceFeedsSchema>[0] | undefined> => {
+  const data = await fetch(`${PUBLIC_URL}/api/pyth/get-feeds/${encodeURIComponent(symbol)}?cluster=${cluster.toString()}`, {
+    next: {
+      revalidate: DEFAULT_CACHE_TTL,
+    },
+    headers: {
+      'x-vercel-protection-bypass': VERCEL_AUTOMATION_BYPASS_SECRET,
+    },
+  });
+  
+  if(!data.ok) {
+    return undefined;
+  }
+
+  const dataJson = await data.text();
+  const feed: z.infer<typeof priceFeedsSchema>[0] = parse(dataJson);
```

Comment:
In general the explicit type declaration is unnecessary and I usually recommend against it, because allowing the type system to automatically unify types is a great tool for refactoring and makes it very easy to find where code actually breaks down -- e.g. if you changed the type of this stuff, this function may actually not need any changes other than type declaration changes, but the actual breaking change would likely be somewhere else in the code.  By writing the explicit declarations and not inferring, you would now have to change these things before you can even find where unification is actually failing.